src/main/java/com/internetitem/logback/elasticsearch/AbstractElasticsearchPublisher.java

```diff
@@ -123,6 +125,20 @@ public void run() {
 
 				List<T> eventsCopy = null;
 				synchronized (lock) {
+					if (outputAggregator.hasPendingData()) {
```

Comment:
We don't add any more events to outputAggregator if it's already choking on data, but retry the request if we still can.